internal/api/router/manager.go

```diff
@@ -67,6 +68,16 @@ func (m *Manager) ListenAndServe(ctx context.Context, address string, port int)
 	router.GET("/v1/report/last/successful", withAuth.Wrap(m.getLastSuccessfulReport))
 	router.POST("/v1/build/trigger", withAuth.Wrap(m.triggerBuild))
 
+	if enablepprof {
```

Comment:
why not using config.Cfg.Debug.Pprof directly and not pass it as param ? in the same way you did line 52 for the authentication.

Reply:
the line 52 is done the wrong way, it should be a param passed to the function to avoid using a global variables.
I'll refactor the line 52 in a dedicated PR

Reply:
Ok, it could be worth to pass a struct (like ManagerConfig) to don't have too many parameters

Reply:
yep I agree

Reply:
for this kind of function I tend to move to struct when reaching 4/5 args minimum